src/dpdk_statefulset_updater.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Module used to update Kubernetes Statefulset to support DPDK."""
+
+import logging
+from typing import Iterable, Optional
+
+from lightkube import Client
+from lightkube.core.exceptions import ApiError
+from lightkube.models.core_v1 import Container
+from lightkube.resources.apps_v1 import StatefulSet
+
+logger = logging.getLogger(__name__)
+
+
+class DPDKStatefulSetUpdaterError(Exception):
+    """DPDKStatefulSetUpdaterError."""
+
+    def __init__(self, message: str):
+        self.message = message
+        super().__init__(self.message)
+
+
+class DPDKStatefulSetUpdater:
+    """Class used to update Kubernetes Statefulset to support DPDK."""
+
+    def __init__(
+        self,
+        statefulset_name: str,
+        namespace: str,
+        dpdk_access_interface_resource_name: str,
+        dpdk_core_interface_resource_name: str,
+    ):
+        self.k8s_client = Client()
+        self.statefulset_name = statefulset_name
+        self.namespace = namespace
+        self.dpdk_resource_requirements = {
+            "requests": {
+                dpdk_access_interface_resource_name: "1",
+                dpdk_core_interface_resource_name: "1",
+            },
+            "limits": {
+                dpdk_access_interface_resource_name: "1",
+                dpdk_core_interface_resource_name: "1",
+            },
+        }
+
+    def container_configured_for_dpdk(self, container_name: str) -> bool:
```

Comment:
```suggestion
    def is_configured(self, container_name: str) -> bool:
```

Reply:
There's already dpdk in the class name, it's self explanatory.

Reply:
Done.